screen-orientation/orientation-reading.html

```diff
@@ -20,6 +22,41 @@
   assert_true(angle == 0 || angle == 90 || angle == 180 || angle == 270);
 }, "Test screen.orientation default values.");
 
+promise_test(async t => {
+  await test_driver.bless("request full screen", () => {
+    return document.documentElement.requestFullscreen();
+  });
+  let primaryOrientation;
+  await screen.orientation.lock("portrait-primary");
+  if (screen.orientation.angle === 0) {
+    primaryOrientation = "portrait-primary";
+  } else {
+    primaryOrientation = "landscape-primary";
+  }
+  if (primaryOrientation === "portrait-primary") {
```

Comment:
Note the code repetition here... and below from line 46... if we look at these two, what we actually are testing is "expected" angles.  

I would test this incrementally and get rid of the if statements - you can then split this into 4 tests. 

```JS
await screen.orientation.lock("portrait-primary");
const {angle, type} = screen.orientation;
assert_equals(type, "portrait-primary");
assert_true(angle === 90 || angle === 270);
```

And so on...